src/Pate/Memory/MemTrace.hs

```diff
@@ -649,40 +652,46 @@ initMemTrace ::
   AddrWidthRepr ptrW ->
   IO (MemTraceImpl sym ptrW)
 initMemTrace sym baseMem Addr32 = do
-  arr <- ioFreshConstant sym "InitMem" knownRepr
+  arrBytes <- ioFreshConstant sym "InitMemBytes" knownRepr
+  arrRegions <- ioFreshConstant sym "InitMemRegions" knownRepr
   sq <- nilSymSequence sym
-  return $ MemTraceImpl sq (MemTraceState arr) (toMuxTree sym Nothing) baseMem
+  return $ MemTraceImpl sq (MemTraceState arrBytes arrRegions) (toMuxTree sym Nothing) baseMem
 initMemTrace sym baseMem Addr64 = do
-  arr <- ioFreshConstant sym "InitMem" knownRepr
+  arrBytes <- ioFreshConstant sym "InitMemBytes" knownRepr
+  arrRegions <- ioFreshConstant sym "InitMemRegions" knownRepr
   sq <- nilSymSequence sym
-  return $ MemTraceImpl sq (MemTraceState arr) (toMuxTree sym Nothing) baseMem
+  return $ MemTraceImpl sq (MemTraceState arrBytes arrRegions) (toMuxTree sym Nothing) baseMem
 
 
 mkMemoryBinding ::
   forall sym ptrW.
+  IsSymExprBuilder sym =>
```

Comment:
I have no objections, but there isn't anything obviously new in this function that might require the additional constraint

Reply:
I just used it to get `OrdF (SymExpr sym)`, which is needed here since we're making a nontrivial Map now.